mesonbuild/backend/ninjabackend.py

```diff
@@ -706,8 +706,9 @@ def generate_compdb(self):
                 # Add custom MIL link rules to get the files compiled by the TASKING compiler family to MIL files included in the database
                 key = OptionKey('b_tasking_mil_link')
                 if key in compiler.base_options:
-                    rules += [f"{rule}{ext}" for rule in [self.get_compiler_rule_name('tasking_mil_compile', compiler.for_machine)]
-                              for ext in ['', '_RSP']]
+                    rule = self.get_compiler_rule_name('tasking_mil_compile', compiler.for_machine)
+                    rules += rule
+                    rules += f'{rule}_RSP'
```

Comment:
```suggestion
                    rules.append(rule)
                    rules.append(f'{rule}_RSP')
```

Unless you want to add each character of the rule as a separate item in the list...